### ISSUE TITLE
Setup travis

### ISSUE THREAD

**Issue by AnEmortalKid** [NexusSandbox/quarut#2]
Travis is nice enough to let OSS libraries build for free. We should take advantage of that for PR bulders.

https://docs.travis-ci.com/user/tutorial/#to-get-started-with-travis-ci

**Comment by NewtrinoPiGui** [comment]
Good idea, though slight complication, Kotlin does not appear to be a supported programming language for their CI: https://docs.travis-ci.com/user/languages/

I believe this can be worked around by bundling the Kotlin SDK with the project, at the expense of a larger package size. I think this is a good option if they do intend on eventually supporting Kotlin, and just haven't gotten around to it yet. I'll do additional research, and possibly log an issue with them if not.

**Comment by NewtrinoPiGui** [comment]
Also, still looking up other options like: https://oss.jfrog.org/artifactory/webapp/#/home

**Comment by AnEmortalKid** [comment]
Could we have the regular java language but the script reference gradle? If we add the gradle wrapper to the repository, we can reference it within Travis as the build script

`./gradlew build`

https://docs.gradle.org/current/userguide/gradle_wrapper.html
